service/tutorial/order/OrderServices.xml

```diff
@@ -0,0 +1,399 @@
+    <service verb="get" noun="ReturnOrderInfo">
+        <in-parameters>
+            <parameter name="orderId"/>
+        </in-parameters>
+        <out-parameters>
+            <parameter name="order" type="List">
+                <parameter name="orderMap" type="Map">
+                    <parameter name="orderId"/>
+                    <parameter name="orderName"/>
+                    <parameter name="currencyUom"/>
+                    <parameter name="salesChannelEnumId"/>
+                    <parameter name="statusId"/>
+                    <parameter name="placedDate"/>
+                    <parameter name="grandTotal"/>
+                    <parameter name="customer_details" type="List">
+                        <parameter name="customerMap" type="Map">
+                            <parameter name="customerPartyId"/>
+                            <parameter name="firstName"/>
+                            <parameter name="middleName"/>
+                            <parameter name="lastName"/>
+                        </parameter>
+                    </parameter>
+                    <parameter name="order_parts" type="List">
+                        <parameter name="partMap" type="Map">
+                            <parameter name="orderPartSeqId"/>
+                            <parameter name="partName"/>
+                            <parameter name="facilityId"/>
+                            <parameter name="shipmentMethodEnumId"/>
+                            <parameter name="partStatusId"/>
+                            <parameter name="partTotal"/>
+                            <parameter name="item_details" type="List">
+                                <parameter name="itemMap" type="Map">
+                                    <parameter name="orderItemSeqId"/>
+                                    <parameter name="productId"/>
+                                    <parameter name="itemDescription"/>
+                                    <parameter name="quantity"/>
+                                    <parameter name="unitAmount"/>
+                                </parameter>
+                            </parameter>
+                        </parameter>
+                    </parameter>
+                </parameter>
+            </parameter>
+        </out-parameters>
+        <actions>
+            <!--1.Fetch data from orderHeader table -->
+            <entity-find-one entity-name="mantle.order.OrderHeader" value-field="order_detail">
+                <field-map field-name="orderId"/>
+            </entity-find-one>
+            <set field="order" from="[]"/>
+            <!--2. fetch data from orderPart -->
+            <entity-find entity-name="mantle.order.OrderPart" list="order_parts">
+                <econdition field-name="orderId"/>
+            </entity-find>
+            <!--3.here setting customerPartyId for use in list customerDetails-->
+            <set field="customerPartyId" from="order_parts.customerPartyId"/>
```

Comment:
Can you check to use .first  with list, In this case, we have same customerPartyId set in the orderParts, but we need to make sure we retrieve the right data from the list.